data/input_files/tests/nonlinear_solid/qs_attribute_solve.lua

```diff
@@ -42,14 +45,20 @@ nonlinear_solid = {
 
     -- boundary condition parameters
     boundary_conds = {
-        {
-            name = "displacement_x",
+        ['displacement_x'] = {
             -- boundary attribute 1 (index 0) is fixed (Dirichlet) in the x direction
             attrs = {1},
+            component = 0,
```

Comment:
Why is the component still needed if the name of the BC is `displacement_x`?

Reply:
I wasn't sure what naming convention we wanted for boundary conditions - right now I'm just checking if the name of the field is in the name of the BC and applying it accordingly.